### PR TITLE
URLRoutingClient updates

### PR DESCRIPTION
**Pull request by stephencelis**
- Rename `URLRoutingClient.request(_:...)` to `decodedResponse(for:)`
- Add `URLRoutingClient.data(for:)`

### CODE REVIEW

**Comment by stephencelis** [comment]
Thanks @Jeehut for the discussion!

File: Sources/URLRouting/Printing.swift [hunk 1]
```diff
@@ -51,7 +51,7 @@ extension ParserPrinter where Input == URLRequestData {
 /// let authenticatedRouter = router
 ///   .baseRequestData(.init(headers: ["X-PointFree-Session": ["deadbeef"]]))
 ///
-/// try authenticatedRouter.request(for: .home)
+/// try authenticatedRouter.response(for: .home)
```
Comment (mbrandonw) [inline Sources/URLRouting/Printing.swift]: should this actually be something more like this?

```suggestion
/// try authenticatedRouter.decodedResponse(for: .home)
///   .response
```

Comment (stephencelis) [reply Sources/URLRouting/Printing.swift]: This was an accidental rename in the docs. It's the API for getting a URLRequest from a router.

Comment (mbrandonw) [reply Sources/URLRouting/Printing.swift]: ah coo

**Review by mbrandonw — APPROVED** [review]

**Pull request merged**